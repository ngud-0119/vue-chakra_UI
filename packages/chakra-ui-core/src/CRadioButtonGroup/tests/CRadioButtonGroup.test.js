import { CButton, CRadioButtonGroup } from '../..'
import { render } from '@/tests/test-utils'

const CustomRadio = {
  name: 'CustomRadio',
  props: {
    isChecked: Boolean,
    isDisabled: Boolean,
    value: [String, Number],
    mx: String,
    dataTestid: String
  },
  inheritAttrs: true,
  render (h) {
    return h(CButton, {
      props: {
        ...this.$props,
        isDisabled: this.isDisabled,
        variantColor: this.isChecked ? 'red' : 'gray'
      },
      attrs: {
        role: 'radio',
        'aria-checked': this.isChecked,
        'data-testid': this.dataTestid
      }
    }, this.$slots.default)
  }
}

const renderComponent = (props) => {
  const base = {
    components: { CRadioButtonGroup, CustomRadio },
    template: `
    <CRadioButtonGroup defaultValue="item-2" isInline>
      <CustomRadio value="item-1" data-testid="item-1">Custom Radio 1</CustomRadio>
      <CustomRadio value="item-2" data-testid="item-2">Custom Radio 2</CustomRadio>
      <CustomRadio value="item-3" data-testid="item-3">Custom Radio 3</CustomRadio>
      <CustomRadio isDisabled value="item-4" data-testid="item-4">Custom Radio 4</CustomRadio>
    </CRadioButtonGroup>
`,
    ...props
  }
  return render(base)
}

it('should render correctly', () => {
  const { asFragment } = renderComponent()

  expect(asFragment()).toMatchSnapshot()
})

it('should display children', () => {
  const { getByText } = renderComponent()
  expect(getByText('Custom Radio 1')).toBeInTheDocument()
  expect(getByText('Custom Radio 2')).toBeInTheDocument()
})

test('defaultValue works', () => {
  const { getByTestId } = renderComponent()
  const one = getByTestId('item-1')
  const two = getByTestId('item-2')

  expect(one).not.toHaveAttribute('aria-checked')
  expect(two).toHaveAttribute('aria-checked', 'true')
})

test('CustomRadio isDisabled works', () => {
  const { getByTestId } = renderComponent()
  const last = getByTestId('item-4')

  expect(last).toHaveAttribute('disabled')
})
