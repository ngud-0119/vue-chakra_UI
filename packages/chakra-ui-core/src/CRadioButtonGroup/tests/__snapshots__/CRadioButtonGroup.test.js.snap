// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  line-height: 1.2;
  outline: 2px solid transparent;
  outline-offset: 2px;
  font-weight: var(--chakra-fontWeights-bold);
  border-radius: var(--chakra-radii-md);
  height: var(--chakra-sizes-10);
  min-width: var(--chakra-sizes-10);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: var(--chakra-colors-red-400);
  color: var(--chakra-colors-white);
}

.emotion-1:focus,
.emotion-1[data-focus] {
  outline: 2px solid transparent;
  outline-offset: 2px;
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-1:hover,
.emotion-1[data-hover] {
  background: var(--chakra-colors-red-500);
}

.emotion-1:active,
.emotion-1[data-active] {
  background: var(--chakra-colors-red-600);
}

.emotion-1[disabled],
.emotion-1[aria-disabled=true],
.emotion-1[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  line-height: 1.2;
  outline: 2px solid transparent;
  outline-offset: 2px;
  font-weight: var(--chakra-fontWeights-bold);
  border-radius: var(--chakra-radii-md);
  height: var(--chakra-sizes-10);
  min-width: var(--chakra-sizes-10);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: var(--chakra-colors-gray-100);
}

.emotion-2:focus,
.emotion-2[data-focus] {
  outline: 2px solid transparent;
  outline-offset: 2px;
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-2:hover,
.emotion-2[data-hover] {
  background: var(--chakra-colors-gray-200);
}

.emotion-2:active,
.emotion-2[data-active] {
  background: var(--chakra-colors-gray-300);
}

.emotion-2[disabled],
.emotion-2[aria-disabled=true],
.emotion-2[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

<div
    class="emotion-0"
    data-chakra-component="CRadioButtonGroup"
    role="radiogroup"
  >
    <button
      aria-checked="true"
      class="emotion-1"
      data-chakra-component="CButton"
      data-testid="item-1"
      role="radio"
      tabindex="0"
      type="button"
    >
      Custom Radio 1
    </button>
    <button
      class="emotion-2"
      data-chakra-component="CButton"
      data-testid="item-2"
      role="radio"
      tabindex="-1"
      type="button"
    >
      Custom Radio 2
    </button>
    <button
      class="emotion-2"
      data-chakra-component="CButton"
      data-testid="item-3"
      role="radio"
      tabindex="-1"
      type="button"
    >
      Custom Radio 3
    </button>
    <button
      aria-disabled="true"
      class="emotion-2"
      data-chakra-component="CButton"
      data-testid="item-4"
      disabled="disabled"
      role="radio"
      tabindex="-1"
      type="button"
    >
      Custom Radio 4
    </button>
  </div>
</DocumentFragment>
`;
